devicetypes/smartthings/zigbee-rgb-bulb.src/zigbee-rgb-bulb.groovy

```diff
@@ -0,0 +1,154 @@
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.color", key: "COLOR_CONTROL") {
+				attributeState "color", action:"color control.setColor"
+			}
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		main(["switch"])
+		details(["switch", "refresh"])
+	}
+}
+
+//Globals
+private getATTRIBUTE_HUE() { 0x0000 }
+private getATTRIBUTE_SATURATION() { 0x0001 }
+private getHUE_COMMAND() { 0x00 }
+private getSATURATION_COMMAND() { 0x03 }
+private getCOLOR_CONTROL_CLUSTER() { 0x0300 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.debug event
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	}
+	else {
+		def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+		def cluster = zigbee.parse(description)
+
+		if (zigbeeMap?.clusterInt == COLOR_CONTROL_CLUSTER) {
+			if(zigbeeMap.attrInt == ATTRIBUTE_HUE){  //Hue Attribute
+				def hueValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 255 * 100)
+				sendEvent(name: "hue", value: hueValue, descriptionText: "Color has changed")
+			}
+			else if(zigbeeMap.attrInt == ATTRIBUTE_SATURATION){ //Saturation Attribute
+				def saturationValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 255 * 100)
+				sendEvent(name: "saturation", value: saturationValue, descriptionText: "Color has changed", displayed: false)
+			}
+		}
+		else if (cluster && cluster.clusterId == 0x0006 && cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00){
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: $cluster"
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+			else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		}
+		else {
+			log.info "DID NOT PARSE MESSAGE for description : $description"
+			log.debug zigbeeMap
+		}
+	}
+}
+
+def on() {
+	zigbee.on()
+}
+
+def off() {
+	zigbee.off()
+}
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	return zigbee.onOffRefresh()
+}
+
+def refresh() {
+	zigbee.onOffRefresh() + zigbee.readAttribute(0x0008, 0x00) + zigbee.readAttribute(0x0300, 0x00) + zigbee.readAttribute(0x0300, ATTRIBUTE_HUE) + zigbee.readAttribute(0x0300, ATTRIBUTE_SATURATION) + zigbee.onOffConfig(0, 300) + zigbee.levelConfig() + zigbee.configureReporting(COLOR_CONTROL_CLUSTER, ATTRIBUTE_HUE, 0x20, 1, 3600, 0x01) + zigbee.configureReporting(COLOR_CONTROL_CLUSTER, ATTRIBUTE_SATURATION, 0x20, 1, 3600, 0x01)
```

Comment:
Could use `zigbee.levelRefresh` instead of `zigbee.readAttribute(0x0008, 0x00)`. `zigbee.readAttribute(0x0300, 0x00)` is the same as `zigbee.readAttribute(0x0300, ATTRIBUTE_HUE)` so the first one should be removed. `COLOR_CONTROL_CLUSTER` is defined so you can use that instead of hardcoding `0x0300.